scripts/run-e2e-tests.sh

```diff
@@ -95,6 +95,7 @@ run() {
     ensure_cluster
     
     build_and_run_tests
+    exit $?
```

Comment:
:+1: this seems like the most likely culprit for the success-on-failure thing we were seeing, right?